R/lime.R

```diff
@@ -160,16 +160,16 @@ select_tree <- function(x, y, weights, n_features) {
 select_f_lp <- function(x, y, weights, n_features) {
   shuffle_order <- sample(length(y)) # glm is sensitive to the order of the examples
   fit <- glmnet(x[shuffle_order,], y[shuffle_order], weights = weights[shuffle_order], alpha = 1, nlambda = 300)
-  # In case that no model with correct n_feature size was found
-  if (all(fit$df != n_features)) {
-    stop(sprintf("No model with %i features found with lasso_path. Try a different method.", n_features))
-  }
   has_value <- apply(coef(fit)[-1, ], 2, function(x) x != 0)
   f_count <- apply(has_value, 2, sum)
   row <- which(f_count >= n_features)[1]
   features <- which(has_value[, row])
+  # In case that no model with correct n_feature size was found
```

Comment:
@thomasp85, as a compromise, I opted to take all the features identified when taking less than `n_features` but then randomly selecting the remaining features from the features identified when taking more than `n_features` not already contained within the first list.

Reply:
I generally eer on the side of returning a simpler model, rather than including a random one... LIME is mainly for explaining a model and including random features seems to run counter to that...

Reply:
Makes sense... I will just take the important features of those that are <= to the number requested and leave the others `NA` instead of randomly selecting one.